api/assessment.go

```diff
@@ -164,11 +177,12 @@ type Assessment struct {
 	Status       string                  `json:"status"`
 	Thresholds   assessment.Thresholds   `json:"thresholds"`
 	RiskMessages assessment.RiskMessages `json:"riskMessages" yaml:"riskMessages"`
+	Required     bool                    `json:"required"`
 }
 
 //
 // With updates the resource with the model.
-func (r *Assessment) With(m *model.Assessment) {
+func (r *Assessment) With(m *model.Assessment, resolver *assessment.QuestionnaireResolver) {
```

Comment:
We've got a convention in the Hub API of creating a method called `WithResolver` to pass the resolver into so that With can continue to just take the model rather than taking an optional parameter. But I think using the resolver here may be redundant anyway based on previous comments.